src/mlpack/methods/ann/gan_impl.hpp

```diff
@@ -124,40 +129,46 @@ double GAN<Model, InitializationRuleType, Noise>::Evaluate(
     const size_t i,
     const size_t /*batchSize*/)
 {
+  std::cout << "Enter GAN Evaluate" << std::endl;
   if (!reset)
     Reset();
 
-  currentInput = this->predictors.unsafe_col(i);
-  currentTarget = this->responses.unsafe_col(i);
+  // Alias here
+  currentInput = this->predictors.cols(i, i + batchSize - 1);
+  currentTarget = this->responses.cols(i, i + batchSize - 1);
 
   discriminator.Forward(std::move(currentInput));
+  std::cout << "Fine?" << std::endl;
   double res = discriminator.outputLayer.Forward(
       std::move(boost::apply_visitor(
           outputParameterVisitor,
           discriminator.network.back())), std::move(currentTarget));
```

Comment:
@zoq I'm having a bit of trouble here. When the above function gets called, the `boost::apply_visitor` routine returns a `1 x 1` object and `currentTarget` is of size `1 x batchSize`, which leads to an error with element-wise multiplication inside `discriminator.outputLayer.Forward` which is the `SigmoidCrossEntropyError::Forward`. Could you help me with this?

Reply:
@zoq I'll think over and try out the second approach, as I think we'll have to change the parameters in all the layers for the first approach to work, which is something that would be quite complicated for a prospective user. I'll let you know.

Reply:
Okay, I can make the necessary modifications if you like; we just have to adjust the conv layers, the rest already works with a ``batchSize > 1``.